tron/utils/scribereader.py

```diff
@@ -0,0 +1,158 @@
+import datetime
+import json
+import logging
+import operator
+import socket
+from functools import lru_cache
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+
+try:
+    from scribereader import scribereader  # type: ignore
+    from clog.readers import StreamTailerSetupError  # type: ignore
+except ImportError:
+    scribereader = None  # sorry folks, you'll need to add your own way to retrieve logs
+
+
+log = logging.getLogger(__name__)
+
+
+@lru_cache(maxsize=1)
+def get_region() -> str:
+    """
+    Discover what region we're running in by reading this information from on-disk facts.
+
+    Yelpers: for more information, see y/habitat
+    """
+    with open("/nail/etc/region") as f:
+        return f.read().strip()
+
+
+@lru_cache(maxsize=1)
+def get_superregion() -> str:
+    """
+    Discover what region we're running in by reading this information from on-disk facts.
+
+    Yelpers: for more information, see y/habitat
+    """
+    with open("/nail/etc/superregion") as f:
+        return f.read().strip()
+
+
+@lru_cache(maxsize=1)
+def get_ecosystem() -> str:
+    """
+    Discover what ecosystem we're running in by reading this information from on-disk facts.
+
+    Yelpers: for more information, see y/habitat
+    """
+    with open("/nail/etc/ecosystem") as f:
+        return f.read().strip()
+
+
+@lru_cache(maxsize=1)
+def get_scribereader_host_and_port() -> Optional[Tuple[str, int]]:
+    try:
+        ecosystem = get_ecosystem()
+        superregion = get_superregion()
+        region = get_region()
+    except OSError:
+        log.warning("Unable to read location mapping files from disk, not returning scribereader host/port")
+        return None
+
+    host, port = scribereader.get_tail_host_and_port(
+        ecosystem=ecosystem if ecosystem != "prod" else None, region=region, superregion=superregion,
```

Comment:
nit: I'd add a comment for why ecosystem should be `None` for prod, since it's not obvious at first glance that prod is different for scribereader.